src/common/sr_data_structs.c

```diff
@@ -561,7 +561,7 @@ sr_free_lock_item(void *lock_item)
 {
     CHECK_NULL_ARG_VOID(lock_item);
     sr_lock_item_t *li = (sr_lock_item_t *) lock_item;
-    free(li->filename);
+    free((void*)li->filename);
```

Comment:
Yes, I want to have the `filename` member of the struct `sr_lock_item_s` `const char *`, not a `char *`.

BTW, assignment of a `const char *` to `char *` is allowed, so freeing a literal constant is possible either way, e.g.:
```c
char *str = "abc";
free(str);
```
compiles just fine.